File: PSArborTouch/Physics/ATEnergy.m
//
//  ATEnergy.m
//  PSArborTouch
//
//  Created by Ed Preston on 19/09/11.
//  Copyright 2011 Preston Software. All rights reserved.
//

#import "ATEnergy.h"


@interface ATEnergy ()
// reserved
@end


@implementation ATEnergy

@synthesize sum     = sum_;
@synthesize max     = max_;
@synthesize mean    = mean_;
@synthesize count   = count_;

- (id) init
{
    self = [super init];
    if (self) {
        sum_    = 0.0;
        max_    = 0.0;
        mean_   = 0.0;
        count_  = 0;
    }
    return self;
}


#pragma mark - Internal Interface


@end

File: PSArborTouchSwift/Physics/ATEnergy.swift
//
//  ATEnergy.swift
//  SystemRig - System Test / Debug Rig
//
//  Created by Ed Preston on 26/09/11.
//  Copyright 2015 Preston Software. All rights reserved.
//
//  Translated to Swift by Arnaud Leene on 03/04/2020.
//  Copyright © 2020 Hovering Above. All rights reserved.
//

import CoreGraphics

public struct ATEnergy {
    
// MARK: - public variables
    
    public var sum = CGFloat.zero
    public var max = CGFloat.zero
    public var mean = CGFloat.zero
    public var count: UInt = 0
    
// MARK: - initializers
    
    public init() { }
    
    public init(sum: CGFloat, max:CGFloat, mean: CGFloat, count: UInt) {
        self.sum = sum
        self.max = max
        self.mean = mean
        self.count = count
    }
    
// MARK: - public functions
    
    public mutating func update(sum: CGFloat, max: CGFloat, count: UInt) {
        self.sum = sum
        self.max = max
        self.count = count
        self.mean = self.sum / CGFloat(self.count)
    }

}
